src/main/scala/nodes/learning/ApproximatePCA.scala

```diff
@@ -0,0 +1,73 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.numerics._
+import breeze.stats._
+import com.github.fommil.netlib.LAPACK._
+import edu.berkeley.cs.amplab.mlmatrix.util.QRUtils
+import org.apache.spark.rdd.RDD
+import org.netlib.util.intW
+import pipelines.Logging
+import workflow.Estimator
+
+/**
+ * Created by tomerk11 on 12/16/15.
+ */
+class ApproximatePCAEstimator(dims: Int, q: Int = 10, p: Int = 5) extends Estimator[DenseVector[Float], DenseVector[Float]] with Logging {
+
+  /**
+   * Adapted from the "PCA2" matlab code given in appendix B of this paper:
+   *    https://www.cs.princeton.edu/picasso/mats/PCA-Tutorial-Intuition_jp.pdf
+   *
+   * @param samples A sample of features to be reduced. Often O(1e6). Logically row-major.
+   * @return A PCA Matrix which will perform dimensionality reduction when applied to a data matrix.
+   */
+  def fit(samples: RDD[DenseVector[Float]]): PCATransformer = {
+    val samps = samples.collect.map(_.toArray)
+    val dataMat: DenseMatrix[Double] = convert(DenseMatrix(samps:_*), Double)
```

Comment:
Hmm I would guess you got this code from `PCAEstimator` which was probably written before we had the `MatrixUtils.rowsToMatrix` helper method. I can change it both here and in `PCAEstimator`